server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -97,6 +97,9 @@
 
     private boolean ccsMinimizeRoundtrips = true;
```

Comment:
Should we switch to a Boolean with a default value of `null` ? This way we can decide the default value depending on whether the minCompatibleShardNode is set or not ? It feels wrong that you have to explicitly disable this option when using minCompatibleShardNode. It should be done automatically unless the user sets the ccsMinimizeRoundtrips value explicitly.

Reply:
@jimczi I've made the change, but slightly different from your suggestion as it wasn't easy to enforce a certain value for `ccsMinimizeRoundtrips` in all cases. Thus, I've removed the setter and kept as the sole option of setting `minCompatibleShardNode` for a search request - a constructor (which will also set `ccsMinimizeRoundtrips`).